crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -945,12 +947,23 @@ def upgrade_first_cluster_node(node, other_node)
         other_node_api.disable_pre_upgrade_attribute_for node.name
         # delete old pacemaker resources (from the node where old pacemaker is running)
         delete_pacemaker_resources other_node
+        # set stateless clones to false if needed after deleting pacemaker resources
+        enable_clone_stateless_services(node)
         # start crowbar-join at the first node
         node_api.post_upgrade
         node_api.join_and_chef
         node_api.save_node_state("controller", "upgraded")
       end
 
+      def enable_clone_stateless_services(node)
+        pacemaker_role_name = node.roles.grep(/^pacemaker-config-.*/).first
+        return if pacemaker_role_name.nil?
+        pacemaker_role = RoleObject.find_role_by_name(pacemaker_role_name)
+        return if pacemaker_role.default_attributes["pacemaker"]["clone_stateless_services"]
+        pacemaker_role.default_attributes["pacemaker"]["clone_stateless_services"] = false
+        pacemaker_role.save
```

Comment:
So if I understand ruby correctly, this is setting 'clone_stateless_services' to false if 'clone_stateless_services' hasn't been defined. In which case isn't it disabling clone stateless services? And only when it isn't defined.
If all that is correct, then maybe the name "disable_clone_stateless_services_by_default" or something, to indicate that it disables, but only if not already defined.

Reply:
Only changes it if its set to true, but I like the name `disable_clone_stateless_services`

Reply:
I am not sure if it is correct ... I have a default SOC7 setup where I did not edit this option, and it `clone_stateless_services` is set to true in pacemaker-config-data role.
Shouldn't we rather change it to false aways?